internal/data/spicedb.go

```diff
@@ -75,6 +76,67 @@ func NewSpiceDbRepository(c *conf.Data, logger log.Logger) (*SpiceDbRepository,
 	return &SpiceDbRepository{client}, cleanup, nil
 }
 
+func (s *SpiceDbRepository) LookupSubjects(ctx context.Context, subject_type, relation string, object *apiV1.ObjectReference, limit uint32, continuation biz.ContinuationToken) (chan *biz.SubjectResult, chan error, error) {
+	var cursor *v1.Cursor = nil
+	if continuation != "" {
+		cursor = &v1.Cursor{
+			Token: string(continuation),
+		}
+	}
+
+	client, err := s.client.LookupSubjects(ctx, &v1.LookupSubjectsRequest{
+		Resource: &v1.ObjectReference{
+			ObjectType: object.Type,
+			ObjectId:   object.Id,
+		},
+		Permission:              relation,
+		SubjectObjectType:       subject_type,
+		OptionalSubjectRelation: "",
+		OptionalConcreteLimit:   limit,
+		OptionalCursor:          cursor,
+	})
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	subjects := make(chan *biz.SubjectResult)
+	errs := make(chan error, 1)
+
+	go func() {
+		for {
+			msg, err := client.Recv()
+			if err != nil {
+				if !errors.Is(err, io.EOF) {
+					errs <- err
+				}
+				close(errs)
+				close(subjects)
+				return
+			}
+
+			continuation := biz.ContinuationToken("")
+			if msg.AfterResultCursor != nil {
+				continuation = biz.ContinuationToken(msg.AfterResultCursor.Token)
+			}
+
+			subj := msg.GetSubject()
+			subjects <- &biz.SubjectResult{
+				Subject: &apiV1.SubjectReference{
+					Object: &apiV1.ObjectReference{
+						Type: subject_type,
+						Id:   subj.SubjectObjectId,
+					},
+					Relation: "",
```

Comment:
Maybe! I realized I wasn't actually taking a subject relation parameter on the request, so it wasn't possible to send queries like `zed permission lookup-subjects group:omni member group#member`, so I added that thinking it would use the relation on output ... though that still wouldn't make a ton of sense since it would always be the same as the relation in the input, and it seems Authzed has thought of that: the ResolvedSubject type doesn't have a relation field on it.

So ya, we may want to do something similar- maybe even flatten out the SubjectResult to just the id since the type is also implied by the request.

Reply:
A potential other thing- SpiceDB's resolved subject type isn't a cut down subject reference, it also supports representing things like subject wildcards (and subject wildcards _except_ excluded subjects) - I don't know if we intend to do anything like that (ie: it would mean having not just user:* as a boolean but also using it as a subject, like in role bindings, say, for operations/resources that are available to all authenticated users in any org), but it is part of SpiceDB's API.

Reply:
Yeah I think we'll probably want to make the response richer, I commented on the proto around this. But it might make sense as a (near-term) follow up PR.

Reply:
Yeah, I think a follow-up makes sense. There's more we can expose here, but I'm not sure how it lines up with our intended use or how it generalizes over Zanzibar implementations, so it seems like more discussion is warranted.